src/tests/Jenkinsfile

```diff
@@ -1,33 +1,64 @@
+properties([
+  pipelineTriggers([[$class: 'PeriodicFolderTrigger', interval: '1d']]),
+  parameters([
+    string(name: 'API_URL', defaultValue: 'https://api.quantum-inspire.com/', description: 'Quantum Inspire API url'),
+    string(name: 'API_CREDENTIALS_ID', defaultValue: 'api.quantum-inspire.com', description: 'Jenkins credentials ID'),
+  ])
+])
 pipeline {
     agent {
         dockerfile {
               filename 'src/tests/Dockerfile.build'
               args '-v pip_cache:/var/pip_cache'
         }
     }
+    environment {
+       PERSIST_WORKSPACE = ""
+    }
     stages {
-        stage('Build') {
+        stage('Checkout modules') {
             steps {
                 sh 'virtualenv venv'
-                sh '. ./venv/bin/activate && pip install .[qiskit,projectq] --cache-dir /var/pip_cache'
-                sh 'pwd'
-                sh 'ls -lah'
-            }
-        }
-        stage('Tests') {
-            steps {
                 sh '''
-                       . ./venv/bin/activate &&
-                       coverage run --source="./src/quantuminspire" -m unittest discover -s src/tests -t src -v &&
-                       coverage report -m &&
-                       coverage xml
+                    . ./venv/bin/activate &&
+                    pwd &&
+                    ls -lah &&
+                    pip install qiskit &&
+                    pip uninstall -y qiskit &&
+                    git clone https://github.com/Qiskit/qiskit &&
+                    pip install ./qiskit -r ./qiskit/requirements-dev.txt
                    '''
             }
         }
-        stage('Coverage') {
+        stage('Build') {
+            steps {
+                sh 'virtualenv venv'
+                sh '. ./venv/bin/activate && pip install . --cache-dir /var/pip_cache -r ./src/tests/requirements.txt'
+           }
+        }
+        stage('Backend Test') {
             steps {
-                step([$class: 'CoberturaPublisher', coberturaReportFile: 'coverage.xml'])
+                script {
```

Comment:
You don't actually need to persist the workspace here, since you are not accessing anything in a subsequent step. It doesn't harm, but if it is not needed better to just remove it to avoid confusion.

Reply:
I'll remove it